pay-api/tests/unit/api/test_payment_request.py

```diff
@@ -881,6 +884,91 @@ def test_payment_request_online_banking_with_credit(session, client, jwt, app):
+                     headers=headers)
+    auth_account_id = rv.json.get('accountId')
+    payment_account: PaymentAccountModel = PaymentAccountModel.find_by_auth_account_id(auth_account_id)
+
+    # Set up EFT credit records
+    eft_file = EFTFileModel()
+    eft_file.file_ref = 'test.txt'
+    eft_file.save()
+
+    eft_credit_1 = EFTCreditModel()
+    eft_credit_1.eft_file_id = eft_file.id
+    eft_credit_1.payment_account_id = payment_account.id
+    eft_credit_1.amount = 50
+    eft_credit_1.remaining_amount = 50
+    eft_credit_1.save()
+
+    eft_credit_2 = EFTCreditModel()
+    eft_credit_2.eft_file_id = eft_file.id
+    eft_credit_2.payment_account_id = payment_account.id
+    eft_credit_2.amount = 45.50
+    eft_credit_2.remaining_amount = 45.50
+    eft_credit_2.save()
+
+    # Create invoice to use credit on
+    token = jwt.create_jwt(get_claims(), token_header)
+    headers = {'Authorization': f'Bearer {token}', 'content-type': 'application/json'}
+    rv = client.post('/api/v1/payment-requests', data=json.dumps(
+        get_payment_request_with_payment_method(business_identifier='CP0002000',
+                                                payment_method=PaymentMethod.EFT.value)),
+                     headers=headers)
+    invoice_id = rv.json.get('id')
+
+    client.patch(f'/api/v1/payment-requests/{invoice_id}?applyCredit=true', data=json.dumps({}), headers=headers)
+    invoice: InvoiceModel = InvoiceModel.find_by_id(invoice_id)
+
+    assert invoice is not None
+    assert invoice.paid == 50
+    assert invoice.total == 50
+    assert invoice.invoice_status_code == InvoiceStatus.PAID.value
+    assert eft_credit_1.remaining_amount == 0
+    assert eft_credit_2.remaining_amount == 45.50
+
+    # Test partial paid with credit
+    headers = {'Authorization': f'Bearer {token}', 'content-type': 'application/json'}
+    rv = client.post('/api/v1/payment-requests', data=json.dumps(
+        get_payment_request_with_payment_method(business_identifier='CP0002000',
+                                                payment_method=PaymentMethod.EFT.value)),
+                     headers=headers)
+    invoice_id = rv.json.get('id')
+
+    client.patch(f'/api/v1/payment-requests/{invoice_id}?applyCredit=true', data=json.dumps({}), headers=headers)
+    invoice: InvoiceModel = InvoiceModel.find_by_id(invoice_id)
+
+    assert invoice is not None
+    assert invoice.paid == 45.50
+    assert invoice.total == 50
+    assert invoice.invoice_status_code == InvoiceStatus.PARTIAL.value
+    assert eft_credit_1.remaining_amount == 0
+    assert eft_credit_2.remaining_amount == 0
+
+    # Increase credit and test for left over balance
+    eft_credit_2.amount = 60
+    eft_credit_2.remaining_amount = 14.50
+    eft_credit_2.save()
+
+    # Apply credit to the previous partial invoice
+    client.patch(f'/api/v1/payment-requests/{invoice_id}?applyCredit=true', data=json.dumps({}), headers=headers)
```

Comment:
I don't think this is needed, I'll have to look at the frontend that uses this

Reply:
Reverted, and moved the test to the payment account service